### PR TITLE
Tempo 2.0: Config Cleanup

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
This is a config cleanup pass to prepare for 2.0. The following changes were made to config:

***Defaults Updated***
Settings listed with new defaults. These defaults were changed to work better with the new Parquet backend.
```yaml
query_frontend:
    max_oustanding_per_tenant: 2000
    search:
        concurrent_jobs: 1000
        target_bytes_per_job: 104857600
        max_duration: 168h
        query_ingesters_until: 30m
    trace_by_id:
        query_shards: 50
querier:
    max_concurrent_queries: 20
    search:
        prefer_self: 10
ingester:
    concurrent_flushes: 4
    max_block_duration: 30m
    max_block_bytes: 524288000
storage:
    trace:
        pool:
            max_workers: 400
            queue_depth: 20000
        search:
            read_buffer_count: 32
            read_buffer_size_bytes: 1048576
```

***Renamed/Moved/Removed Config***
```yaml
query_frontend:
    query_shards: // removed. use trace_by_id.query_shards
querier:
    query_timeout: // removed. use trace_by_id.query_timeout
compactor:
    compaction:
        chunk_size_bytes:   // renamed to v2_in_buffer_bytes
        flush_size_bytes:     // renamed to v2_out_buffer_bytes
        iterator_buffer_size: // renamed to v2_prefetch_traces_count
ingester:
    use_flatbuffer_search:   // removed. automatically set based on block type
storage:
    wal:
        encoding:  // renamed to v2_encoding
        version:     // removed and pinned to block.version
    block:
        index_downsample_bytes:  // renamed to v2_index_downsample_bytes
        index_page_size_bytes:      // renamed to v2_index_page_size_bytes
        encoding:                  	    // renamed to v2_encoding
        row_group_size_bytes:       // renamed to parquet_row_group_size_bytes
```

As part of the removing `useFlatBufferSearch` and requiring the wal and block `Version` to be the same the tests in `./modules/ingester` had to be heavily reworked. This is b/c they were written for flatbuffer/v2 wal and they have now been updated to use the current block type for the wal (vparquet). During this process the following bugs were found and corrected:

- SearchTags/TagValues was not working on the parquet headblock/completing blocks
- When replaying a parquet wal block we were double counting total objects

Finally, I would still like to remove `search_enabled` and `metrics_generator_enabled` but I intend to do that in a future PR as this has gotten quite large.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by electron0zero — APPROVED** [review]

File: tempodb/encoding/common/config.go [hunk 1]
```diff
@@ -8,17 +8,19 @@ import (
...
+	BloomFP             float64          `yaml:"bloom_filter_false_positive"`
+	BloomShardSizeBytes int              `yaml:"bloom_filter_shard_size_bytes"`
+	Version             string           `yaml:"version"`
+	SearchEncoding      backend.Encoding `yaml:"search_encoding"`
+	SearchPageSizeBytes int              `yaml:"search_page_size_bytes"`
+
+	// v2 fields
+	IndexDownsampleBytes int              `yaml:"v2_index_downsample_bytes"`
+	IndexPageSizeBytes   int              `yaml:"v2_index_page_size_bytes"`
+	Encoding             backend.Encoding `yaml:"v2_encoding"`
```
Comment (mapno) [inline tempodb/encoding/common/config.go:20]: We should warn if these fields are configured and version is not v2.

**Review by mapno — APPROVED** [review]

File: modules/storage/config.go [hunk 2]
```diff
@@ -37,7 +36,6 @@ func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet)
 
 	cfg.Trace.WAL = &wal.Config{}
 	f.StringVar(&cfg.Trace.WAL.Filepath, util.PrefixConfig(prefix, "trace.wal.path"), "/var/tempo/wal", "Path at which store WAL blocks.")
-	cfg.Trace.WAL.Version = v2.VersionString
```
Comment (mdisibio) [inline modules/storage/config.go:40]: The ability to configure the wal separately from the backend was useful for iteration and experimentation.  What is the rationale to remove it?

Comment (joe-elliott) [reply modules/storage/config.go:40]: Agreed. I have restored `wal.version` which defaults to `block.version` if there's an empty string. It is purposefully undocumented.

File: tempodb/encoding/common/config.go (hunk 1 above)
Comment (joe-elliott) [reply tempodb/encoding/common/config.go:20]: +1 added

File: tempodb/wal/wal.go [hunk 3]
```diff
@@ -30,15 +30,17 @@ type Config struct {
...
-	Version           string           `yaml:"version"`
 	IngestionSlack    time.Duration    `yaml:"ingestion_time_range_slack"`
+	Version           string           `yaml:"version,omitempty"`
 }
 
-func ValidateConfig(b *Config) error {
-	if _, err := encoding.FromVersion(b.Version); err != nil {
-		return err
+// jpe test me
+// jpe test at the ingester level that defaults work
```
Comment (mdisibio) [inline tempodb/wal/wal.go:40]: remaining todo?

**Review by mdisibio — APPROVED** [review]

File: tempodb/wal/wal.go (hunk 3 above)
Comment (joe-elliott) [reply tempodb/wal/wal.go]: good catch. removed

**Pull request merged**